### PR TITLE
Implement Import Model Function

### PR DESCRIPTION
**Pull request by TruongQuangSB**
In this PR:
- Implemenet import model function in development mode.
- At this moment the retarget reference ID function isn't working.
- After modification of model and retarget reference ID working should the import function in normal mode avaible

### CODE REVIEW

**Review by pstuecker — APPROVED** [review]

File: java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+ */
+public class ImportControl {
+	/**
+	 * Target to import of the control
+	 */
+	public enum ImportTarget {
+		/**
+		 * Import whole sub work (include INITIAL and FINAL container)
+		 */
+		SUBWORK,
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java:63]: ALL / FULL?

File: java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+		 * Import to current initial container of selected sub work
+		 */
+		INITIAL,
+		/**
+		 * Import to current final container of selected sub work
+		 */
+		FINAL
+	}
+
+	PlanPro_Schnittstelle schnittstelle;
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java:75]: Should be private

File: java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+	ImportComboFileField comboField;
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java:76]: Should be private

File: java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+	private Function<ImportComboFileField, Boolean> comboValidCheck;
+	private boolean isValid = false;
+	Option option;
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java:80]: Should be private

File: java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Shell;
+
+/**
+ * Control for import data from anothe project
+ * 
+ * @author Truong
+ */
+public class ImportControl {
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/src/org/eclipse/set/feature/projectdata/ppimport/ImportControl.java:55]: The control severely mixes logic with UI elements. Also the importing logic is separated into various files with a very unclear data flow (Parts in ImportControl, ModelImportUtils/PlanProImportPart).

**Review by pstuecker — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/constants/ValidationResult.java [hunk 6]
```diff
@@ -49,6 +49,26 @@ public static enum Outcome {
 		VALID
 	}
 
+	/**
+	 * Validate state of file
+	 */
+	public static enum FileValidateState {
```
[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/constants/ValidationResult.java:55]: u are right, they are same

File: java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties [hunk 7]
```diff
@@ -14,7 +14,8 @@ PlanProImportPart_infoGroup=Informationen zur geöffneten PlanPro-Datei
 PlanProImportPart_countStart=Anzahl der Objekte im Startzustand
 PlanProImportPart_countZiel=Anzahl der Objekte im Zielzustand
 PlanProImportPart_notSupportedInfo=Der Import von Fachdaten wird nur für LST-Planungen unterstützt.
-PlanProImportPart_importGroup=Import von Fachdaten
+PlanProImportPart_importGroup=Import Daten von andere PlanPro-Datei
+PlanProImportPart_importSubwork=PlanPro-Datei für Import neu Fachdaten auswählen
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties:18]:
```suggestion
PlanProImportPart_importSubwork=PlanPro-Datei für Import von Fachdaten auswählen
```

File: java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties [hunk 8]
```diff
@@ -23,7 +24,12 @@ PlanProImportPart_errorMsg=Die Datei konnte nicht geladen werden.
 PlanProImportPart_infoErrorCause=Ursache
 PlanProImportPart_incompletePlanProFile=Unvollständige PlanPro-Datei
 PlanProImportPart_incompletePlanProFileMessage=Achtung, die Datei "%s" ist für den Anwendungsfall PT 1 nicht vollständig. Wenn Sie die Datei trotzdem importieren, so kann es zu fehlerhaften oder unvollständigen Darstellungen kommen.
-
+PlanproImportPart_Subwork_Notset=Fehl Untergewerk Art
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties:27]:
```suggestion
PlanproImportPart_Subwork_Notset=Es wurde kein Untergewerk gewählt
```

File: java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties [hunk 9]
```diff
@@ -23,7 +24,12 @@ PlanProImportPart_errorMsg=Die Datei konnte nicht geladen werden.
...
+PlanProImportPart_Subwork_NotSelected=Quellfachdaten wählen
+PlanProImportPart_Subwork_All=Alle Fachdaten
+PlanProImportPart_SelectSubworkMessage=Ausgewählt Fachdaten zu importieren enthält keine Untergewerk_Art. Bitte wählen eine Art oder ändern Fachdaten
+PlanProImportPart_SelectSubworkLabel=Untergewerk Art
+PlanProImportPart_ImportSameData=Sie haben dasselbe ausgewählen, Fachdaten oder Container zu importieren.Bitte ändern Sie die Auswahl.
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.feature.projectdata/OSGI-INF/l10n/bundle.properties:32]:
```suggestion
PlanProImportPart_ImportSameData=Es kann nur ein Untergewerk pro Untergewerksart importiert werden.
```

**Review by pstuecker — CHANGES_REQUESTED** [review]
- Changing the file after selecting a subwork allows pressing the import button and causing an exception.

**Review by pstuecker — APPROVED** [review]

**Pull request merged**